loky/backend/resource_tracker.py

```diff
@@ -212,13 +237,16 @@ def main(fd, verbose=0):
         sys.stderr.write("Main resource tracker is running\n")
         sys.stderr.flush()
 
-    cache = {rtype: set() for rtype in _CLEANUP_FUNCS.keys()}
+    cache = {rtype: dict() for rtype in _CLEANUP_FUNCS.keys()}
     try:
         # keep track of registered/unregistered resources
         if sys.platform == "win32":
             fd = msvcrt.open_osfhandle(fd, os.O_RDONLY)
         with open(fd, 'rb') as f:
-            for line in f:
+            while True:
+                line = f.readline()
+                if line == b'':  # EOF
+                    break
```

Comment:
Question: why no for loop anymore?

Reply:
Using a for-loop, in `Python 2`, the `resource_tracker` would wait until it got `EOF` before iterating over the lines written inside the pipe, processing them all at once (during shutdown). This is a bug that is actually present in `loky` master (but only for `Python 2`)